FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_ARP.c

```diff
@@ -114,47 +118,47 @@ uint32_t ulTargetProtocolAddress, ulSenderProtocolAddress;
 	pxARPHeader = &( pxARPFrame->xARPHeader );
 
 	/* The field ulSenderProtocolAddress is badly aligned, copy byte-by-byte. */
-	memcpy( ( void *)&( ulSenderProtocolAddress ), ( void * )pxARPHeader->ucSenderProtocolAddress, sizeof( ulSenderProtocolAddress ) );
+	( void ) memcpy( ( void * )&( ulSenderProtocolAddress ), ( const void * ) pxARPHeader->ucSenderProtocolAddress, sizeof( ulSenderProtocolAddress ) );
 	/* The field ulTargetProtocolAddress is well-aligned, a 32-bits copy. */
 	ulTargetProtocolAddress = pxARPHeader->ulTargetProtocolAddress;
 
 	traceARP_PACKET_RECEIVED();
 
 	/* Don't do anything if the local IP address is zero because
-	that means a DHCP request has not completed. */
+	 * that means a DHCP request has not completed. */
 	if( *ipLOCAL_IP_ADDRESS_POINTER != 0UL )
 	{
 		switch( pxARPHeader->usOperation )
 		{
-			case ipARP_REQUEST	:
+			case ipARP_REQUEST :
 				/* The packet contained an ARP request.  Was it for the IP
-				address of the node running this code? */
+				 * address of the node running this code? */
 				if( ulTargetProtocolAddress == *ipLOCAL_IP_ADDRESS_POINTER )
 				{
 					iptraceSENDING_ARP_REPLY( ulSenderProtocolAddress );
 
 					/* The request is for the address of this node.  Add the
-					entry into the ARP cache, or refresh the entry if it
-					already exists. */
+					 * entry into the ARP cache, or refresh the entry if it
+					 * already exists. */
 					vARPRefreshCacheEntry( &( pxARPHeader->xSenderHardwareAddress ), ulSenderProtocolAddress );
 
 					/* Generate a reply payload in the same buffer. */
-					pxARPHeader->usOperation = ( uint16_t ) ipARP_REPLY;
+					pxARPHeader->usOperation = ( uint16_t )ipARP_REPLY;
 					if( ulTargetProtocolAddress == ulSenderProtocolAddress )
 					{
 						/* A double IP address is detected! */
 						/* Give the sources MAC address the value of the broadcast address, will be swapped later */
-						memcpy( pxARPFrame->xEthernetHeader.xSourceAddress.ucBytes, xBroadcastMACAddress.ucBytes, sizeof( xBroadcastMACAddress ) );
-						memset( pxARPHeader->xTargetHardwareAddress.ucBytes, '\0', sizeof( MACAddress_t ) );
+						( void ) memcpy( ( void * ) pxARPFrame->xEthernetHeader.xSourceAddress.ucBytes, ( const void * ) xBroadcastMACAddress.ucBytes, sizeof( xBroadcastMACAddress ) );
+						( void ) memset( pxARPHeader->xTargetHardwareAddress.ucBytes, 0, sizeof( MACAddress_t ) );
```

Comment:
Why no `( const void * )` on argument 2 here?  You have it everywhere else.

Reply:
Ah, because memset has this signature:
'void *memset(void *str, int c, size_t n)'

Reply:
Sorry I missed that, will resolve